### PR TITLE
newline_chr is optionally user-specified

### PR DESCRIPTION
**Pull request by fisher-j**
R on cmd.exe on windows adds an extra newline using the current logic (newline is chosen as: "\r\n"). This proposal makes `newline_chr` a property of the `Terminal`, and allows the user to specify it's value when creating custom terminals.

### CODE REVIEW

File: lua/toggleterm/terminal.lua [hunk 1]
```diff
@@ -287,10 +290,11 @@ end
 
 ---Combine arguments into strings separated by new lines
 ---@vararg string
+---@param newline_chr string
 ---@return string
-local function with_cr(...)
+local function with_cr(newline_chr, ...)
   local result = {}
-  local newline_chr = get_newline_chr()
+  -- local newline_chr = get_newline_chr()
```
Comment (akinsho) [inline lua/toggleterm/terminal.lua:297]: should this be commented out? or removed?

Comment (fisher-j) [reply lua/toggleterm/terminal.lua:297]: I think this can safely be removed.

File: doc/toggleterm.txt [hunk 2]
```diff
@@ -477,13 +477,18 @@ using the current count e.g. `:5ToggleTerm<CR>`
     local lazygit = Terminal:new({ cmd = "lazygit", count = 5 })
 <
 
-You can also set a custom layout for a terminal.
+You can also set several options for the custom terminal including layout,
+opening and closing functions, and the newline charcter used when sending text
```
Comment (akinsho) [inline doc/toggleterm.txt:481]:
```suggestion
opening and closing functions, and the newline character used when sending text
```

File: doc/toggleterm.txt [hunk 3]
```diff
@@ -477,13 +477,18 @@ using the current count e.g. `:5ToggleTerm<CR>`
...
+to the terminal. The `newline_chr` defaults to "\r" on powershell, "\r\n" on
+cmd.exe, and "\n" otherwise.
 
 >lua
     local lazygit = Terminal:new({
       cmd = "lazygit",
       dir = "git_dir",
       direction = "float",
+      -- Try changing if you get too many/few blank lines when sending to terminal
```
Comment (akinsho) [inline doc/toggleterm.txt:490]: I don't think we need a special addition for this especially given the explainer above

**Review by akinsho — COMMENTED** [review]
LGTM after the last couple of tweaks

File: doc/toggleterm.txt (hunk 3 above)
Comment (fisher-j) [reply doc/toggleterm.txt]: removed

**Review by akinsho — APPROVED** [review]
LGTM, here's hoping nothing breaks

**Pull request merged**